### DISCUSSION TITLE
Popup: Local backup clarity suggestions

### DISCUSSION THREAD

**Discussion by marcustyphoon** [AprilSylph/XKit-Rewritten#144] [Suggestion Box]
Super minor, but it occurred to me that `XKit Payload [number].json` and `Paste your saved export here, then hit the "Restore" button` would be a bit clearer if they matched. Perhaps "export" or "backup" in the file name instead of "Payload" would make a bit more sense.

Additionally, I can easily imagine a user not understanding what to do with the file. `Paste the contents of your saved export here` could be a bit more handholdy, as well as saving the file as .txt instead of .json, which I imagine may not open in a text editor on some computers. Assuming that pasting into a text box to restore is intended as the long term method, this seems to make more sense to me. Naturally, if the import were changed to a file selection interface, .json makes perfect sense.

**Comment by AprilSylph** [comment]
The whole backup interface definitely needs an overhaul. I could also change the differentiator from a unix timestamp to a roughly formatted datetime. However, file uploads directly into a browser action popup doesn't work, as opening the file upload prompt closes the popup, on Firefox at least. Maybe that's been changed in FF89 and it's feasible now since the entire display engine has been switched over, but I'd have to test that.

**Reply by AprilSylph** [reply]
implemented the filename change in 47c3487ea099d72f81f6a979b98255124e74c563

(was hoping mentioning the discussion in the commit would add a mention of it here automatically like in issues, doesn't seem to work that way :pensive:)

**Reply by AprilSylph** [reply]
Sadly, file uploads remain impossible from the browser action popup in Firefox ([Bug 1474423](https://bugzilla.mozilla.org/show_bug.cgi?id=1474423)). However, I have taken the time to overhaul the import/export UI to hopefully make things clearer; both the file produced by the Download button and the Local Import subsection refer to the file as a "backup", and there are now two textareas rather than having the one with two purposes. I've also made what I consider to be visual improvements. There's still some work to be done here, such as making a new interface for cloud storage, and rethinking `storageLastModified`, but please try the current changes out and let me know what you think!

**Reply by marcustyphoon** [reply]
Looking great to me so far! Nice touch on the one-click-to-select-all :D
